### DISCUSSION TITLE
Question about frames' parent joint

### DISCUSSION THREAD

**Discussion by xiaosuyang** [stack-of-tasks/pinocchio#2174] [Q&A] [answered]
Hello.
Using the example urdf ur5 robot and Print the parent joint and previous frame of each joint:
![image](https://github.com/stack-of-tasks/pinocchio/assets/55594017/c2425812-8a5c-451f-ae80-8c6d8174e9e4)
Why some joints' parent joint is itself but some joints not:
![image](https://github.com/stack-of-tasks/pinocchio/assets/55594017/b1f0b40a-aefc-4048-9a8a-71cdd71a0e58)
![image](https://github.com/stack-of-tasks/pinocchio/assets/55594017/1802ac5c-09b9-4d48-93e9-da58a2c19ee3)

**Comment by xiaosuyang** [comment]
It seems that if the joint type is revolute,the parent joint is itself. If the joint type is "fixed",the parent joint is its predecessor

**Comment by MegMll** [comment, marked as answer]
Hello @xiaosuyang, 

In pinocchio, "fixed" joint refers to a frame and not to real joint of the kinematic tree which is why the parent joint of the frame "ee_fixed_joint' is the previous joint that appeared in the tree, ie "wrist_3" and not itself.
